src/main/scheduler/scheduler.c

```diff
@@ -350,79 +375,143 @@ static void readSchedulerLocals(task_t *selectedTask, uint8_t selectedTaskDynami
 
 FAST_CODE void scheduler(void)
 {
-#if defined(USE_LATE_TASK_STATISTICS)
-    static task_t *lastTask;
-#endif
     // Cache currentTime
     const timeUs_t schedulerStartTimeUs = micros();
-    timeUs_t currentTimeUs = schedulerStartTimeUs;
+    uint32_t nowCycles;
     timeUs_t taskExecutionTimeUs = 0;
     task_t *selectedTask = NULL;
     uint16_t selectedTaskDynamicPriority = 0;
     uint16_t waitingTasks = 0;
-    bool realtimeTaskRan = false;
-    timeDelta_t gyroTaskDelayUs = 0;
+    uint32_t nextTargetCycles = 0;
+    int32_t schedLoopRemainingCycles;
 
     if (gyroEnabled) {
         // Realtime gyro/filtering/PID tasks get complete priority
         task_t *gyroTask = getTask(TASK_GYRO);
```

Comment:
The gyroTask, filter task and pid task addresses never change after fc init, so perhaps schedulerInit should likely cache them to avoid getting them every iteration?  See lines 432 and 435 which get the filter and pid tasks.

Reply:
This is a fairly trivial operation.

```
task_t *getTask(unsigned taskId)
{
    return &tasks[taskId];
}
```

I'd argue that we'd be better off saving the little bit of memory that this would consume.

Reply:
yeah, I looked at the implementation too. I would argue though, due to the fact that the scheduler runs more times per second than any other code in the system, that it is the one place where as is optimised as much as possible.

Reply:
@hydra: What optimisation settings did you use when you tested this and it did not get inlined?

Reply:
It did get in-lined if I recall correctly, maybe no need to inline that one, a closer look at the generated assembly for both approaches is probably required.